integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/WaitTerminationChecker.java

```diff
@@ -64,30 +63,14 @@ public String successMessage(T waitObject) {
 
     @Override
     public boolean exitWaiting(T waitObject) {
-        try {
-            waitObject.fetchData();
-            if (waitObject.isDeleteFailed()) {
-                return false;
-            }
-            return waitObject.isFailed();
-        } catch (ProcessingException clientException) {
-            LOGGER.error("Exit waiting! Failed to get cluster due to API client exception: {}", clientException.getMessage(), clientException);
-        } catch (Exception e) {
-            LOGGER.warn("Exit waiting! Failed to get cluster, because of: {}", e.getMessage(), e);
-            return true;
+        if (waitObject.isDeleteFailed()) {
+            return false;
         }
-        return false;
+        return waitObject.isFailed();
     }
 
     @Override
     public Map<String, String> getStatuses(T waitObject) {
-        String name = waitObject.getName();
-        try {
-            waitObject.fetchData();
-            return waitObject.actualStatuses();
-        } catch (NotFoundException e) {
-            LOGGER.warn("No cluster found with name '{}'! It has been deleted successfully.", name, e);
-            return Map.of("status", DELETE_COMPLETED.name());
-        }
+        return deleted ? waitObject.getDesiredStatuses() : waitObject.actualStatuses();
```

Comment:
what will happen in the test if this method gives back the desires statuses? In the catch clause, we used the `DELETE_COMPLETED`